### ISSUE TITLE
[FEATURE] Output hostname from btp_subaccount_environment_instance

### ISSUE THREAD

**Issue by SeanKilleen** [SAP/terraform-provider-btp#455]
### What area do you want to see improved?

terraform provider

### Is your feature request related to a problem? Please describe.

We create a `btp_subaccount_environment_instance` resource as part of creating our Kyma envronment.

Later in our configuration we create an `APIRule` in Kyma via our kubernetes config. Part of that rule requires the current hostname.

As far as I can tell, I can't get to the hostname through kubernetes itself. But I did notice that the `btp_subaccount_environment_instance` data source exports the dashboard URL, which makes me think it might be possible to export the hostname as well, which would be very useful to use downstream in some of that Kubernetes configuration.

Currently, we have to undertake a manual step of copying that hostname into a variable, which it would be lovely to avoid.

### Describe the solution you would like

I'd love for the `btp_subaccount_environment_instance` resource (or data source) to output the hostname of the kyma environment, so that we could use it downstream.

### Describe alternatives you have considered

Getting this from Kyma itself via the kubernetes provider. However, I was not able to find out how to do it (though I certainly could be missing it!)

**Comment by lechnerc77** [comment]
@SeanKilleen Thanks for the feedback. The API we use does not explicitly return this as a parameter, that is why we did not add this parameter up to now. As you mentioned the dashboard URL could be used to fetch the information. 

I would have expected a Terraform function to easily export this but according to this [discussion](https://github.com/hashicorp/terraform/issues/23893) it is not. The discussion highlights that the `regex` functionality can be used to overcome this.

My question: would the following usage of the regex function in your Terraform setup mitigate the manual copy and pasting:

```hcl
regex("^(?:(?P<scheme>[^:/?#]+):)?(?://(?P<authority>[^/?#]*))?", "https://www.terraform.io/docs/")
```

which returns:
```hcl
{
  "authority" = "www.terraform.io"
  "scheme" = "https"
}
```

Although the `authority` can differ from the `hostname`according to the RFC <https://datatracker.ietf.org/doc/html/rfc3986#appendix-B>, with the URL in the dashboard this would be fine. 
 
Does this resolve your issue?

**Comment by SeanKilleen** [comment]
@lechnerc77 sorry, I know I've owed you an answer on this one! 

I realized that the dashboard URL may not actually work, because at present, that URL is actually along the lines of `https://dashboard.kyma.cloud.sap/?kubeconfigID=REDACTED_KUBECONFIG_ID`. As such, the hostname we'd need, e.g. `abc1234.kyma.ondemand.com`, isn't actually in the URL and so can't be parsed from it.

So, while it doesn't solve my issue, the issue is a small one that only occurs on environment setup and then I have a variable for it anyway, so I'm comfortable to have this closed out. I did want to get you the answer though!
